### DISCUSSION TITLE
What happened with Budibase CLI? 'Check for updates' is broken

### DISCUSSION THREAD

**Discussion by marceldarvas** [Budibase/budibase#5021] [Help]
About two months ago, I followed your documentation on Docker deployment via the Budibase CLI. 
At the time, your docs suggested being the "recommended" method when self-hosting via Docker.

Now I've found myself being stuck on version 1.0.46. When I hit the 'Check for updates' button:
![ScreenShot 2022-03-22 at 16 53 09](https://user-images.githubusercontent.com/10864884/159574285-80f61499-1773-4e38-8aec-949b5e4d6bf8.png)

I'm told I'm already on the latest version, which I am not.

When I search for the documentation, I'm referring to [the page is no longer available](https://docs.budibase.com/self-hosting/self-host/budibase-cli-setup)

So what is the recommended approach to fixing an issue like this?

I found the CLI command: `budi hosting --update`

Which then asks me if I want to update "you docker-compose.yaml and envoy.yaml?" - well, I went with No due to the lack of documentation. But it would be great to know if I should fix that.

My install says it's at 1.0.90 now, but it was weird how I ended up in this situation. Please note that I am using CF Tunnel with domain, and the server firewall seems to be okay with accessing external sites for OpenID and SQL. However, the update mechanism was acting up...

I did find [Whitelisting](https://docs.budibase.com/docs/whitelisting) doc with an IP, it doesn't seem like matters in my situation, maybe with the Budibase API coming soon!

**Comment by joebudi** [comment]
Apologies, the CLI setup docs were moved:
https://docs.budibase.com/docs/budibase-cli-setup

**Reply by marceldarvas** [reply]
Thank you and sorry, my bad! Excellent work with changing to Readme.io. It's a cool framework. I am excited about the API Reference!
Maybe a little Regex would help so old doc URLs get sent to new pages.

Just a suggestion: As I was looking in the [Self-Hosting Settings](https://docs.budibase.com/docs/hosting-settings), maybe that category (Self-Hosting) could be changed into Architecture and moving [Architecture](https://docs.budibase.com/docs/budibase-architecture) and [Deployment Information](https://docs.budibase.com/docs/deployment-information) into there.

Reverse Proxy and Hosting Settings could go under [Install](https://docs.budibase.com/docs/hosting-methods) since they point to the deployment methods like the [Docker](https://docs.budibase.com/docs/docker-compose) page, which was the one I was looking at when opening my post but didn't see the Budi CLI mentioned (the previous version reminded people of CLI is the recommended approach for production I believe)

**Comment by mike12345567** [comment]
Hi @marceldarvas - from time to time you may need to update the CLI, either by updating it via npm or downloading one of the latest releases. That's likely the issue you are seeing with it trying to download files that aren't used/don't exist anymore.

Release here: https://github.com/Budibase/budibase/releases/tag/v1.0.90

**Reply by marceldarvas** [reply]
Yes, I see that the release version is now at v1.0.90 and I was able to update via `budi hosting --update` before finishing my post.

 When you referred to "update the CLI" I looked into how to do that, so I ran the following commands:
```
$ npm update
$ npm update budi
$ npm -v budi
```
I didn't get any returns or errors, the last one printed `6.14.4` which confused me with what version numbering it's following...

But thank you for your answers, the one thing that I am unsure about is when I run: `budi hosting --update`
I get `? Do you wish to update you docker-compose.yaml and envoy.yaml? No`

I'm not fully sure what it would need to update in those files, that's why I went with no, but it would be great if you could clarify!

Also: The [Budibase IP](https://docs.budibase.com/docs/whitelisting#budibase-ip) is only for Data Access?
It has nothing to do with "Budibase analytics enabled"?

I thought I'd also mention this bug I've noticed on the builder portal, with the scrollbar appearing instead of wrapping around options:
![ScreenShot 2022-03-23 at 11 16 58](https://user-images.githubusercontent.com/10864884/159737035-0ff9a320-2135-4857-84f1-73f89aeca912.png)
